src/Framework/AspNetCoreAnalyzers/src/SourceGenerators/PublicTopLevelProgramGenerator.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.SourceGenerators;
+
+[Generator]
+public class PublicProgramSourceGenerator : IIncrementalGenerator
+{
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var internalGeneratedProgramClass = context.CompilationProvider.Select((compilation, cancellationToken) =>
+        {
+            var program = compilation.GetTypeByMetadataName("Program");
```

Comment:
This "Program" is the well-known name of the entrypoint class when using top-level statements (as opposed to an aspnetcore concept)?

Reply:
Yep, it's behavior that's guaranteed by the language/compiler (see [this](https://github.com/dotnet/csharplang/blob/66a3a0320fa567b31b37b9045c424c5254da56de/proposals/csharp-9.0/top-level-statements.md#semantics)).